src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindIntegrationTest.java

```diff
@@ -393,6 +393,102 @@ public void inConditionEmptyArray() {
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.documents", hasSize(0))
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()));
+    }
+
+    @Test
+    public void nonIDInConditionEmptyArrayAnd() {
+      String json =
+          """
+                        {
+                          "find": {
+                              "filter" : {
+                                "$and": [
+                                    {
+                                        "age": {
+                                            "$in": []
+                                        }
+                                    },
+                                    {
+                                        "username": "user1"
+                                    }
+                                ]
+                              }
+                            }
+                        }
+                      """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.documents", hasSize(0))
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()));
+    }
+
+    @Test
+    public void nonIDInConditionEmptyArrayOr() {
+      String json =
+          """
+                        {
+                          "find": {
+                              "filter" : {
+                                "$or": [
+                                    {
+                                        "age": {
+                                            "$in": []
+                                        }
+                                    },
+                                    {
+                                        "username": "user1"
+                                    }
+                                ]
+                              }
+                            }
+                        }
+                      """;
+      String expected1 =
+          "{\"_id\":\"doc1\", \"username\":\"user1\", \"active_user\":true, \"date\" : {\"$date\": 1672531200000}}";
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.documents", hasSize(1))
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents[0]", jsonEquals(expected1));
```

Comment:
It might be better to have a test case that returns no results here, because there's no way of telling whether the document `expected1` matched the first condition in `$or`, the 2nd, or both.